### PR TITLE
Migrate users db

### PR DESCRIPTION
**Pull request by dannyroberts**
@snopoke 

The bulk migrate command was stupid fast on my local machine (300 users in no perceivable time/under a second); hopefully it's not too bad on prod, probably ~1h.

### CODE REVIEW

**Comment by czue** [comment]
build fail real?
```
======================================================================
ERROR: test_number_of_calls (corehq.util.tests.test_couch.IterDBSimpleTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/travis/build/dimagi/commcare-hq/corehq/util/tests/test_couch.py", line 101, in test_number_of_calls
    iter_db.save(doc)
  File "/home/travis/build/dimagi/commcare-hq/corehq/util/couch.py", line 142, in __exit__
    self.commit()
  File "/home/travis/build/dimagi/commcare-hq/corehq/util/couch.py", line 137, in commit
    self._commit_save()
  File "/home/travis/build/dimagi/commcare-hq/corehq/util/couch.py", line 116, in _commit_save
    success_ids = self._write(bulk_save, self.to_save)
  File "/home/travis/build/dimagi/commcare-hq/corehq/util/couch.py", line 101, in _write
    results = cmd(docs)
TypeError: bulk_save() got an unexpected keyword argument 'new_edits'
```

File: corehq/doctypemigrations/migrator.py [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+        self.record_seq(self.source_db.info()['committed_update_seq'])
+        bulk_migrate(self.source_db, self.target_db, self.doc_types,
+                     filename=self.data_dump_filename)
+
+    def phase_2_replicate_continuously(self):
+        pass
+
+    def record_seq(self, seq):
+        state, _ = DocTypeMigrationState.objects.get_or_create(slug=self.slug)
+        state.original_seq = seq
```
Comment (snopoke) [inline corehq/doctypemigrations/migrator.py]: Shouldn't there be a `state.save()` call here?

**Comment by dannyroberts** [comment]
YAAAAASSS @snopoke I think good to merge now. cc: @czue

**Comment by czue** [comment]
nice! what's the plan for keeping things in sync after the migration?

**Comment by czue** [comment]
(nevermind found it)

**Comment by esoergel** [comment]
@dannyroberts These migrations are blocking even on the first run of `./manage.py migrate` after setting up a new dev environment.  Any thoughts on how to make that a noop?  Maybe check to see if there are any matching docs or something.

**Comment by snopoke** [comment]
@millerdev did something like that in the blobdb migration. I think if
there are less than 500 docs it just does the migration directly.

**Comment by dannyroberts** [comment]
Yeah I like the idea of just doing it if there are only a few docs, even though that's technically not 100% safe in a production setting, I can't imagine it every conceivably causing any problems. Skipping if there are no docs to migrate is 100% safe and would probably also get the job done.

**Comment by esoergel** [comment]
I opened [this PR](https://github.com/dimagi/commcare-hq/pull/9894) adopting the convention @gcapalbo proposed [here](https://github.com/dimagi/commcare-hq/blob/f320562823b153548f3e85d9be85057fd286e9d9/corehq/apps/sms/migration_status.py#L61), which special-cases fresh installs.

**Comment by millerdev** [comment]
> that's technically not 100% safe in a production setting

The management command should always be run manually in a production setting prior to when the django migration is deployed.

**Pull request merged**